### PR TITLE
Fixed double underline

### PR DESCRIPTION
**Pull request by lllong33**
resolves https://github.com/dbt-labs/dbt-core/issues/5301

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [x] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

image

### CODE REVIEW

**Comment by mikealfare** [comment]
This macro is no longer used in `dbt-redshift` (in favor of a redshift-specific one). So we don't need to make any updates there. I agree with @jtcohen6's concern in the attached issue, that we need to maintain the name and point to the updated macro for backwards compatibility, so I'd like to see that in this PR as well.

**Comment by lllong33** [comment]
@mikealfare thx for review, are the examples of backwards compatibility?

**Comment by lllong33** [comment]
image

**Review by mikealfare — CHANGES_REQUESTED** [review]

**Comment by mikealfare** [comment]
Thanks for the updates @lllong33. I had a chance to read through this more thoroughly and provided some guidance. What we're effectively doing here is generalizing an adapter-specific macro. There are implementations of this in other adapters, so that makes sense. As a summary, here's the general pattern we use for doing such a thing:
```sql
-- in dbt-core
{% macro my_awesome_macro(arg1, arg2, ...) %}
    -- implement anything we want to happen all of the time, such as logging (we don't do that in this case)
    -- dispatch to the adapter-specific version
{% endmacro %}

{% macro default__my_awesome_macro(arg1, arg2, ...) %}
    -- provide a default in case the macro is not implemented for the adapter
    -- throw an exception to indicate that the macro is not implemented for the adapter unless there is a very universal default, but we tend to not want to assume anything to reduce coupling between adapters
{% endmacro %}

-- in dbt-postgres (or whatever adapter you're working on)
{% postgres__my_awesome_macro(arg1, arg2, ...) %}
    -- the adapter-specific implementation of `my_awesome_macro`
{% endmacro %}
```
The last macro is usually the only one that has code in it, the others are really just for opening an interface in the adapter framework.

**Comment by lllong33** [comment]
Many thanks to @mikealfare for patient guidance.
Here's my latest tweak, but in the get_relations example in test_macro. I can't get the adapter postgres__get_relatins correctly.
My operating steps is to modify the DBT-core code, then run `make dev`, then run `dbt compile -s test_macro`
image

File: core/dbt/include/global_project/macros/adapters/metadata.sql [hunk 1]
```diff
@@ -63,3 +63,12 @@
   {{ exceptions.raise_not_implemented(
     'list_relations_without_caching macro not implemented for adapter '+adapter.type()) }}
 {% endmacro %}
+
+{% macro get_relations() %}
+  {{ return(adapter.dispatch('get_relations')) }}
```
Comment (mikealfare) [inline core/dbt/include/global_project/macros/adapters/metadata.sql:68]: This is missing the 'dbt' parameter and it's not actually calling the macro, it's just passing it in. I think this line needs to look like this:
```sql
    {{ return(adapter.dispatch('get_relations', 'dbt')()) }}
```

**Comment by lllong33** [comment]
good ! 
image

**Review by mikealfare — APPROVED** [review]

**Pull request merged**